svgdigitizer/electrochemistry/cv.py

```diff
@@ -19,110 +19,199 @@
-            df_['I_A'] = df['I']/1E6
+        df_ = df.copy()
+        q = 1 * CV.get_axis_unit(self.svgplot.units['y'])
 
-        df_['I_mA'] = df_['I_A']*1E3
-        df_['I_uA'] = df_['I_A']*1E6
+        # Distinguish whether the y data is current ('A') or current density ('A / cm2')
+        if 'm2' in str(q.unit):
+            conversion_factor = q.to(u.A / u.m**2)
+        else:
+            conversion_factor = q.to(u.A)
 
-        return df_[['I_A', 'I_mA', 'I_uA']]
+        df_[self.axis_properties['y']['dimension']] = df_['y'] * conversion_factor
 
-    def create_df_time_axis(self, df):
+        return df_
+
+    def _add_time_axis(self, df):
         r'''
-        Create a time axis in the dataframe based on the scan rate given in the
-        figure description.
+        Add time column to the dataframe `df`, based on the :meth:`rate`.
         '''
-        df_ = df.copy()
-        df_['deltaU'] = abs(df_[self.xlabel].diff())
-        df_['cumdeltaU'] = df_['deltaU'].cumsum()
-        df_['t'] = df_['cumdeltaU']/self.get_rate()
-        return df_['t']
-
-    def plot_cv(self):
-        self.df.plot(x='U_V', y='I_uA')
-        plt.xlabel(f'{self.xlabel} / {self.xunit}')
-        plt.ylabel(f'{self.ylabel} / {self.yunit}')
-
-    def create_csv(self, filename):
-        csvfile = Path(filename).with_suffix('.csv')
-        self.df.to_csv(csvfile, index=False)
+        df = df.copy()
+        df['deltaU'] = abs(df['U'].diff().fillna(0))
+        df['cumdeltaU'] = df['deltaU'].cumsum()
+        df['t'] = df['cumdeltaU'] / float(self.rate.to(u.V / u.s).value)
+
+        return df
+
+    def plot(self):
+        self.cv_df.plot(x=self.axis_properties['x']['dimension'], y=self.axis_properties['y']['dimension'])
+        plt.axhline(linewidth=1, linestyle=':', alpha=0.5)
+        plt.xlabel(self.axis_properties['x']['dimension'] + ' / ' + str(self.axis_properties['x']['unit']))
+        plt.ylabel(self.axis_properties['y']['dimension'] + ' / ' + str(self.axis_properties['y']['unit']))
+
+    @property
+    def metadata(self, comment=''):
+        metadata = self._metadata.copy()
+        metadata.setdefault('figure description', {})
+        metadata['figure description']['type'] = 'digitized'
+        metadata['figure description']['measurement type'] = 'CV'
+        metadata['figure description']['scan rate'] = {'value': self.rate.value, 'unit': str(self.rate.unit)}
+        metadata['figure description'].setdefault('potential scale', {})
+        metadata['figure description']['potential scale']['unit'] = str(CV.get_axis_unit(self.svgplot.units['x']))
+        # TODO: Get the reference from the text labels of the axis #59
+        # such as: metadata['figure description']['potential scale']['reference'] = self.axis_properties['x']['reference']
+        metadata['figure description']['current'] = {'unit': str(CV.get_axis_unit(self.svgplot.units['y']))}
+        metadata['figure description']['comment'] = str(comment)
```

Comment:
```suggestion
        if comment: metadata['figure description']['comment'] = str(comment)
```
Do we want a "comment" even if there's none?

Reply:
It could probably be omitted. I also think that this field should be added to the yaml file, when the svg is created. In that case, it should not even appear in this `metadata`..
The problem is related to the standard yaml file below, which should probably be fixed separately. 
For the time being, I suggest removing `comment` as an argument and also remove this line.